Allow use of glibc/libintl on GNU/Hurd

pinotree opened the pull request:
GNU/Hurd uses GNU libc, which ships libintl; hence, when building using a system gettext installation on the Hurd, there is nothing else to do.

Advertize this new support in the READMEs of the crates.

Minoru commented:
Looks good, thank you!

Please update READMEs for gettext-sys and gettext-rs crates -- they have lists of OSes where `gettext-system` attempts to use the built-in implementation.

Minoru commented:
Don't mind the macOS 12 checks, apparently it's time to disable them as GitHub is retiring the runner images.

Minoru commented:
I fixed macOS checks on the master branch, so feel free to rebase onto it.

pinotree committed:
Allow use of glibc/libintl on GNU/Hurd

GNU/Hurd uses GNU libc, which ships libintl; hence, when building using
a system gettext installation on the Hurd, there is nothing else to do.

Advertize this new support in the READMEs of the crates.

Minoru commented:
Thank you very much!
